<?xml version="1.0"?>
<!-- Generic launch file for ROSBuzz  -->
<!-- This file is included in all ROS workspace launch files  -->
<!-- Modify with great care!  -->
<launch>
  <arg name="name" default="robot0"/>
  <arg name="xbee_plugged" default="true"/>
  <arg name="script" default="main"/>
  <arg name="wpfile" default="waypointlist"/>
  <arg name="launch_config" default="topics"/>
  <arg name="debug" default="false" />
  <arg name="setmode" default="false" />
  <arg name="latitude" default="0" />
  <arg name="longitude" default="0" />
  <arg name="ca_on" default="2" />
  <arg name="auto_state" default="WAYPOINT" />

  <node name="rosbuzz_node" pkg="rosbuzz" type="rosbuzz_node" respawn="false" output="screen" >
    <rosparam file="$(find rosbuzz)/launch/launch_config/$(arg launch_config).yaml"/>
    <param name="bzzfile_name" value="$(find rosbuzz)/buzz_scripts/$(arg script).bzz" />
    <param name="WPfile" value="$(find rosbuzz)/buzz_scripts/include/taskallocate/$(arg wpfile).csv" />
    <param name="debug" value="$(arg debug)" />
    <param name="xbee_plugged" value="$(arg xbee_plugged)"/>
    <param name="name" value="$(arg name)"/>
    <param name="setmode" value="$(arg setmode)"/>
    <param name="latitude" value="$(arg latitude)"/>
    <param name="longitude" value="$(arg longitude)"/>
    <param name="stand_by" value="$(find rosbuzz)/buzz_scripts/stand_by.bzz"/>
    <param name="ca_on" value="$(arg ca_on)"/>
    <param name="auto_state" value="$(arg auto_state)"/>
  </node>
</launch>
